lib/generator/helpers.dart

```diff
@@ -23,3 +25,16 @@ Iterable<T> mergeDuplicatesBy<T, U>(Iterable<T> list,
   });
   return values.values.toList();
 }
+
+/// checks if the passed queries contain at least one non nullable field
+bool hasNonNullableInput(Iterable<QueryDefinition> queries) {
+  for (var query in queries) {
+    for (var input in query.inputs) {
```

Comment:
you can also use [Iterable.any](https://api.flutter.dev/flutter/dart-core/Iterable/any.html), but it's the same thing